src/SharpCompress/Archives/AbstractWritableArchive.cs

```diff
@@ -96,10 +97,22 @@ IArchiveEntry IWritableArchive.AddEntry(string key, Stream source, bool closeStr
             }
             var entry = CreateEntry(key, source, size, modified, closeStream);
             newEntries.Add(entry);
-            RebuildModifiedCollection();
+            if(shouldRebuildModifiedCollection)
+                RebuildModifiedCollection();
             return entry;
         }
 
+        public void PauseInternalEntryUpdates()
```

Comment:
I'd make a IDisposable object that would flip the boolean back to true on Dispose so that way a using statement can be used in the implementation

Reply:
I thought about that. Such a pause-object would need to know the instance of the AbstractWritableArchive in order to flip the boolean and to call RebuildModifiedCollection on that instance when the pause-object is disposed.
This requires the user to use the using statement - if the user don't and the user doesn't dispose the pause-object himself, then it will be disposed some time later where the AbstractWritableArchive  might have been disposed. It is difficult to get right.
Therefore I opted for the pattern used in WinForms where SuspendLayout/ResumeLayout is commonly used for performance reasons when larger updates of lists or tree structures are updated.

So I don't see how to do this in a nice way (but I am happy to be educated on the issue ;-) if you can show me how)